test/test_rack_env.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'helpers/svr_in_proc'
+
+# Since there isn't a way to pull the env passed to the app,
+# one of the apps defined in svr_in_proc.rb (rack_env_to_body)
+# returns the env as a response body
+
+class TestRackEnv < ::TestPuma::SvrInProc
+  parallelize_me!
+
+  def test_rack_url_scheme_http
+    # default bind is http
+    rack_url_scheme 'http'
+  end
+
+  def test_rack_url_scheme_https
+    skip_unless :ssl
+    bind_type :ssl
+    rack_url_scheme 'https'
+  end
+
+  def test_rack_url_scheme_user
+    bind_type config: "rack_url_scheme 'user'"
+    rack_url_scheme 'user'
+  end
+
+  def test_rack_url_scheme_aunix_http
+    skip 'Waiting for abstract unix sockets'
+    skip_unless :aunix
+    bind_type :aunix
+    rack_url_scheme 'http'
+  end
+
+  private
+
+  def rack_url_scheme(val)
+    start_server rack_env_to_body
+    assert_includes fast_connect_get_body.lines, "rack.url_scheme: #{val}\n"
```

Comment:
We should have a higher-level assert:

```
assert_body_includes "rack.url_scheme"
```

...so test writers don't have to understand `fast_connect_get_body.lines`

Reply:
Done